cel/program.go

```diff
@@ -108,6 +136,15 @@ func newProgram(e *env, ast Ast, opts ...ProgramOption) (Program, error) {
 		}
 	}
 
+	// Configure the resolver to support unknown attributes if the eval option is set.
+	resolver := interpreter.NewResolver(e.adapter)
+	if p.evalOpts&OptUnknownAttributes == OptUnknownAttributes {
+		resolver = interpreter.NewUnknownResolver(resolver)
+	}
+	interp := interpreter.NewInterpreter(
```

Comment:
Nit: inline `p.interpreter`?

Reply:
I've refactored the code here a bit to reduce the boiler-plate. Hopefully it looks better now.